packages/graphiql/src/utility/QueryStore.js

```diff
@@ -64,13 +65,38 @@
   }
 
   push(item) {
-    this.items.push(item);
-    this.save();
-  }
+    const items = [...this.items, item];
+
+    if (this.maxSize && items.length > this.maxSize) {
+      items.shift();
+    }
+
+    let done = false;
+    let success = false;
+    let tries = 0;
+    while (!done) {
```

Comment:
Can we use a for loop here to make the termination a little clearer? i.e.
```
for(let i = 0; i < 5; i++) {
  const response = this.storage.set(this.key, JSON.stringify({ [this.key]: items }));;
  if (response.error === null) {
    this.items = items;
  } else if (response.quotaError) { // Why are we nullchecking `maxSize` here? I think we can skip that check?
    items.shift();
  } else {
    return; // We don't know what happened in this case, so just bailing out
  }
}

Reply:
Null check is for favorites store. I'm using maxSize as a proxy for knowing if I'm in a LRU (queries) or a persistent store (favorites). I'll update the code.